python/feathub/processors/flink/table_builder/flink_table_builder.py

```diff
@@ -408,6 +407,30 @@ def _get_table_from_derived_feature_view(
             right_table = table_by_names[right_table_name]
             right_table_descriptor = self.registry.get_features(right_table_name)
             redis_source = get_redis_source(right_table_descriptor)
+            right_table = optimize_redis_source_lookup_map(
+                self.t_env,
+                right_table,
+                right_table_descriptor,
+                right_table_join_field_descriptors,
+            )
+
+            select_dict = {}
+            for field_name in right_table.get_schema().get_field_names():
+                select_dict[field_name] = f"`{field_name}`"
```

Comment:
Instead of enumerating all existing columns of the `right_table`, would it be simpler to only add the columns for those `right_table_join_field_descriptors` that contain "static map lookup op"? We can add columns similar to what is done in `append_physical_key_columns_per_feature()`.

And would it be more readable to move this logic into `join_utils#lookup_join`?

Reply:
I have moved this logic to `redis_utils` instead of `join_utils` to avoid circular reference.